server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -352,7 +380,11 @@ public void doRun()
           {
             try {
               for (Pair<FireHydrant, Interval> pair : indexesToPersist) {
-                metrics.incrementRowOutputCount(persistHydrant(pair.lhs, schema, pair.rhs));
+                metrics.incrementRowOutputCount(
```

Comment:
hmmm... that sounds correct, still thinking what would be the right thing to do here...
may be create a marker file in the end at /persist_dir/datasource/<SUCCESS>, and use commit metadata information only if the marker file was present?